src/main/java/de/tum/in/www1/artemis/web/rest/StaticCodeAnalysisResource.java

```diff
@@ -105,13 +99,46 @@ public ResponseEntity<Set<StaticCodeAnalysisCategory>> resetStaticCodeAnalysisCa
         log.debug("REST request to reset static code analysis categories for programming exercise {}", exerciseId);
 
         ProgrammingExercise programmingExercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
+        checkSCAEnabledForExerciseElseThrow(programmingExercise);
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, programmingExercise, null);
+
+        Set<StaticCodeAnalysisCategory> staticCodeAnalysisCategories = staticCodeAnalysisService.resetCategories(programmingExercise);
+        return ResponseEntity.ok(staticCodeAnalysisCategories);
+    }
 
+    /**
+     * PATCH /programming-exercises/:exerciseId/static-code-analysis-categories/import
+     *
+     * @param exerciseId       The exercise to copy the configuration into
+     * @param sourceExerciseId The exercise to take the existing configuration from
+     * @return The newly created SCA configuration
+     * @see StaticCodeAnalysisService#importCategoriesFromExercise(ProgrammingExercise, ProgrammingExercise)
+     */
+    @PatchMapping(Endpoints.IMPORT)
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<Set<StaticCodeAnalysisCategory>> importStaticCodeAnalysisCategoriesFromExercise(@PathVariable Long exerciseId, @RequestParam Long sourceExerciseId) {
+        log.debug("REST request to import static code analysis categories to programming exercise {} from exercise {}", exerciseId, sourceExerciseId);
+
+        ProgrammingExercise targetExercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
+        checkSCAEnabledForExerciseElseThrow(targetExercise);
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, targetExercise, null);
+
+        ProgrammingExercise sourceExercise = programmingExerciseRepository.findByIdElseThrow(sourceExerciseId);
+        checkSCAEnabledForExerciseElseThrow(sourceExercise);
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, sourceExercise, null);
```

Comment:
Optional: You loaded the user earlier for two of these checks but not here.

Reply:
this code is not part of my PR, so I'm not sure if I should change anything here.